src/Config.php

```diff
@@ -142,6 +144,24 @@ public function registerCustomFixers(iterable $fixers): ConfigInterface
         return $this;
     }
 
+    /**
+     * Registers custom rule sets to be used the same way as built-in rule sets.
+     *
+     * `$ruleSets` must follow `'@RuleName' => RuleSetDescriptionInterface::class` convention.
+     *
+     * @param array<string, class-string<RuleSetDescriptionInterface>> $ruleSets
+     *
+     * @todo Introduce it in ConfigInterface in 4.0
```

Comment:
👍🏻

Reply:
myself, I have a belief that only interfaces should be exposed as contracts.

Reply:
Myself, I have strong belief that introducing interface for anything that has only one possible implementation does not make sense. It's over-engineering and shot in the foot, because you can't freely extend that contract. It's totally OK to make final class as a contract as long as you keep backward compatibility of the API. Having such class as a contract means that you can introduce new methods at any point, and they will instantly have general availability for the users.

Reply:
Not to mention that with suffix-less naming you can introduce interface at any point in place of the class, while renaming implementation (e.g. `class Foo` becomes `class DefaultFoo implements Foo` and `interface Foo` is introduced - all signatures and the usage is intact, but now it become flexible and can be implemented differently by the users). That's why I don't like `Interface` suffix 😉 (well, not _only_ because of that).

Reply:
It was reworked and now this TODO is in `CustomRulesetsAwareConfigInterface::registerCustomRuleSets()`. There's no BC-break, other config implementations can satisfy this contract too as an opt-in.